### PR TITLE
[PVR][addons] PVR API 5.5.0: Add support for series link based series recordings, ...

### PR DESCRIPTION
**Pull request by ksooo**
This is another PR for a PVR Addon API change for Leia. It contains both the API changes and the related Kodi core changes.

1. New timer type attribute PVR_TIMER_TYPE_REQUIRES_EPG_SERIESLINK_ON_CREATE
================================================================
Todo for addon maintainers: none

2. New field PVR_TIMER::strSeriesLink
===========================================================
Todo for addon maintainers: none

3. New field EPG_TAG::strSeriesLink
===========================================================
Todo for addon maintainers: none

4. Removed unused API function GetChannelSwitchDelay
===========================================================
Todo for addon maintainers: Cleanup your code by removing this function.

5. New API function GetStreamTimes
===========================================================
Refer to #12598 for details.  This PR will be merged the same time as this one here.
Todo for addon maintainers: Add minimalistic support for this function (return PVR_ERROR_NOT_IMPLEMENTED;)

Functional changes in Kodi: If a timer rule shall be created from an epg tag and this tag contains a serieslink, this serieslink will be placed into the new timer and transferred to the client addon; so, the addon can create a series recording based on that serieslink. Additionally, the addon can define timer types requiring a series link in an epg. Using this, it is possible for the addon to provide for example a "Series link" timer type which appears in the timer settings dialog:

28745069-2f5aa378-7470-11e7-8817-24a483a6f949
28745070-2f7d87f8-7470-11e7-96a4-ef9693be2ad5

This change has been runtime tested on macOS, latest kodi master.

No addon code changes required, a simple recompile is sufficient.

Reference implementation for pvr.hts: https://github.com/kodi-pvr/pvr.hts/pull/321

### CODE REVIEW

**Comment by ksooo** [comment]
@Jalle19 for review?

**Review by Jalle19 — APPROVED** [review]
Apart from the comment, +1

File: xbmc/pvr/epg/EpgInfoTag.h [hunk 1]
```diff
@@ -280,6 +280,12 @@ namespace PVR
     int SeriesNumber(void) const;
 
     /*!
+     * @brief The series link for this event.
+     * @return The series link or empty string, if not available.
+     */
+    std::string SeriesLink() const;
```
Comment (ksooo) [inline xbmc/pvr/epg/EpgInfoTag.h]: I just followed the style of the other getters for this class. Wanted to have it consistent at least.

Comment (Jalle19) [inline xbmc/pvr/epg/EpgInfoTag.h]: Okay

**Pull request merged**